test/jdk/java/net/httpclient/DebugLoggerTest.java

```diff
@@ -0,0 +1,321 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.ByteArrayOutputStream;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.logging.Handler;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import jdk.internal.net.http.common.Utils;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/*
+ * @test
+ * @summary Verify the behaviour of the debug logger.
+ * @library /test/lib /test/jdk/java/net/httpclient/lib
+ * @build jdk.httpclient.test.lib.common.HttpServerAdapters jdk.test.lib.net.SimpleSSLContext
+ *        DebugLoggerTest
+ * @run main/othervm  DebugLoggerTest
+ * @run main/othervm -Djdk.internal.httpclient.debug=errr  DebugLoggerTest
+ * @run main/othervm -Djdk.internal.httpclient.debug=err  DebugLoggerTest ERR
+ * @run main/othervm -Djdk.internal.httpclient.debug=out  DebugLoggerTest OUT
+ * @run main/othervm -Djdk.internal.httpclient.debug=log  DebugLoggerTest LOG
+ * @run main/othervm -Djdk.internal.httpclient.debug=true DebugLoggerTest ERR LOG
+ * @run main/othervm -Djdk.internal.httpclient.debug=err,OUT  DebugLoggerTest ERR OUT
+ * @run main/othervm -Djdk.internal.httpclient.debug=err,out,log DebugLoggerTest ERR OUT LOG
+ * @run main/othervm -Djdk.internal.httpclient.debug=true,log DebugLoggerTest ERR LOG
+ * @run main/othervm -Djdk.internal.httpclient.debug=true,out DebugLoggerTest ERR OUT LOG
+ * @run main/othervm -Djdk.internal.httpclient.debug=err,OUT,foo  DebugLoggerTest ERR OUT
+ */
+public class DebugLoggerTest {
+    static final PrintStream stdErr = System.err;
+    static final PrintStream stdOut = System.out;
+    static final String LOGGER_NAME = "jdk.internal.httpclient.debug";
+    static final String MESSAGE = "May the luck of the Irish be with you!";
+    static final String MESSAGE2 = "May the wind be at your back!";
+    static final String MESSAGE3 = "May the sun shine warm upon your face!";
+
+    static RecordingPrintStream err = null;
+    static RecordingPrintStream out = null;
+
+    /**
+     * A {@code RecordingPrintStream} is a {@link PrintStream} that makes
+     * it possible to record part of the data stream in memory while still
+     * forwarding everything to a delegated {@link OutputStream}.
+     * @apiNote
+     * For instance, a {@code RecordingPrintStream} might be used as an
+     * interceptor to record anything printed on {@code }System.err}
```

Comment:
Typo, should have been `{@code System.err}` instead of `{@code } System.err}`

Reply:
Good catch